lib/src/facts/solaris/operating_system_resolver.cc

```diff
@@ -54,7 +54,7 @@ namespace facter { namespace facts { namespace solaris {
          to be resolved further using the `pkg info kernel` command (TODO).
          */
 
-        static boost::regex regexp_s10("Solaris \\d+ \\d+/\\d+ s(\\d+)x_u(\\d+)wos_");
+        static boost::regex regexp_s10("Solaris \\d+ \\d+/\\d+ s(\\d+)\\S_u(\\d+)wos_");
```

Comment:
I lean towards being slightly more restrictive and following Facter 2's example with `[sx]` rather than `\\S`.